app/plugin_requests.go

```diff
@@ -40,6 +41,29 @@ func (a *App) ServePluginRequest(w http.ResponseWriter, r *http.Request) {
 	a.servePluginRequest(w, r, hooks.ServeHTTP)
 }
 
+// ServePluginStaticRequest serves static plugin files
+// at the URL http(s)://$SITE_URL/plugins/$PLUGIN_ID/public/{anything}
+func (a *App) ServePluginStaticRequest(w http.ResponseWriter, r *http.Request) {
+	if strings.HasSuffix(r.URL.Path, "/") {
+		http.NotFound(w, r)
+		return
+	}
+
+	// Should be in the form of /$PLUGIN_ID/public/{anything} by the timne we get here
+	vars := mux.Vars(r)
+	pluginID := vars["plugin_id"]
+	staticfile := vars["static_file"]
```

Comment:
Can `static_file` end up as `../../config.json` and allow exposing files outside the plugin bundle? Minimally, we need a test here to verify, but I'm wondering if it would be more obvious to instead examine `r.URL.Path` directly:
```go
publicFilePath := path.Clean(r.URL.Path)
prefix := fmt.Sprintf("%s/public/", pluginID)
if !strings.HasPrefix(publicFilePath, prefix) {
	http.NotFound(w, r)
	return
}

publicFilePath = filepath.Join(publicFilesPath, strings.TrimPrefix(publicFilePath, prefix));
```

Reply:
I agree. I had this in a previous commit for the same concerns you had. I think I lost that check when I simplified this code in later commits. I'll add it back in.